sycl/test/warnings/warnings.cpp

```diff
@@ -1,20 +1,17 @@
-// RUN: %clangxx -fsycl --no-system-header-prefix=CL/ -Wall -Wextra -Wno-ignored-attributes -Wno-deprecated-declarations -Wpessimizing-move -Wunused-variable -Wmismatched-tags -Wunneeded-internal-declaration -Werror -Wno-unknown-cuda-version %s -o %t.out
+// RUN: %clangxx -fsycl --no-system-header-prefix=CL/sycl -fsyntax-only -Wall -Wextra -Wno-ignored-attributes -Wno-deprecated-declarations -Wpessimizing-move -Wunused-variable -Wmismatched-tags -Wunneeded-internal-declaration -Werror -Wno-unknown-cuda-version %s -o %t.out
 
 #include <CL/sycl.hpp>
 
 using namespace cl::sycl;
-
-int main(void) {
-  // add a very simple kernel to see if compilation succeeds with -Werror
-  int data1[10] = {-1, -1, -1, -1, -1, -1, -1, -1, -1, -1};
-  buffer<int, 1> B(data1, range<1>(10), {property::buffer::use_host_ptr()});
-  queue Q;
-  Q.submit([&](handler &CGH) {
-    auto Accessor = B.get_access<access::mode::read_write>(CGH);
-    CGH.parallel_for<class TheSimpleKernel>(range<1>{10}, [=](id<1> Item) {
-      Accessor[Item] = 0;
+int main() {
+  vec<long, 4> newVec;
+  queue myQueue;
+  buffer<vec<long, 4>, 1> resultBuf {&newVec, range<1> {1}};
+  myQueue.submit([&](handler& cgh) {
+    auto writeResult = resultBuf.get_access<access::mode::write>(cgh);
+    cgh.single_task<class kernel_name>([=]() {
+      writeResult[0] = (vec<int, 4>{1, 2, 3, 4}).template convert<long>();
     });
   });
```

Comment:
Are there any cases that were covered with old test and not covered by the new test?

Reply:
I think no, there was just a simple program that launches the check of warnings.

Reply:
But maybe we should ask the code owner.